### PR TITLE
Add AnechoicRoom class

### PR DESCRIPTION
**Pull request by duembgen**
This pull request reopens #160, now against the new version of pyroomacoustics. 

This will close issue #13.

I am working on polishing it and will remove the "WIP" tag when done. 

- [x] Are there docstrings ? Do they follow the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) style ?
- [x] Have you run the tests by doing `nosetests` or `py.test` at the root of the repo ?
- [x] Have you checked that the doc builds properly and that any new file has been added to the repo ? How to do that is covered in the [documentation](https://pyroomacoustics.readthedocs.io/en/pypi-release/contributing.html#documentation).
- [x] Is there a unit test for the proposed code modification ? If the PR addresses an issue, the test should make sure the issue is fixed.
- [x] Last but not least, did you document the proposed change in the CHANGELOG file ? It should go under "Unreleased".

### CODE REVIEW

**Comment by fakufaku** [comment]
Hi @duembgen ! What is the status on this pull request ? If it is good, I could just pull it in. The tests are failing now, so maybe a merge or something is needed.

File: README.rst [hunk 1]
```diff
@@ -158,7 +158,7 @@ The minimal dependencies are::
 
 where ``Cython`` is only needed to benefit from the compiled accelerated simulator.
 The simulator itself has a pure Python counterpart, so that this requirement could
-be ignored, but is much slower.
+be ignored, but is much slower. For the Cython implementation to work, the user should have Eigen installed.
```
Comment (fakufaku) [inline README.rst:161]: Actually, if you check the repo with the `--recursive` option, it should pull the eigen library too. I don't think the note is needed. This is covered in `CONTRIBUTING.rst`.

**Review by fakufaku — COMMENTED** [review]
I could not find the definition of the `AnechoicRoom` anymore ? Is the code in sync ?

**Comment by duembgen** [comment]
Hi @fakufaku thank you for your comments! I am finalizing the pull request now. Will remove the Eigen comment if it is fixed by doing a recursive clone. Do you have any preference on how / if to mention the AnechoicRoom class in the documentation? I added another section called Free-field simulation in the room doc file, but happy to change that.

File: pyroomacoustics/room.py [hunk 2]
```diff
@@ -2871,3 +2869,122 @@ def is_inside(self, pos):
...
+        The relative humidity of the air in the room (between 0 and 100). By
+        default set to 0.
+    air_absorption: bool, optional
+        If set to True, absorption of sound energy by the air will be
+        simulated.
+    """
+
+    def __init__(
+        self,
+        dim,
```
Comment (fakufaku) [inline pyroomacoustics/room.py:2909]: Can we make this `dim=3` by default ? I would like to move away from 2D room simulation in the long term :)

File: pyroomacoustics/room.py [hunk 3]
```diff
@@ -2871,3 +2869,122 @@ def is_inside(self, pos):
...
+        )
+
+    def __str__(self):
+
+        return "AnechoicRoom instance in {}D.".format(self.dim)
+
+    def is_inside(self, p):
+        """Overloaded function to eliminate testing if objects are inside ``room''."""
+        # always return True because we want the walls to have no effect.
+        return True
```
Comment (fakufaku) [inline pyroomacoustics/room.py:2961]: That's an interesting way to get around the size problem 🙈 I had not thought of that... But I like it!

File: pyroomacoustics/room.py [hunk 4]
```diff
@@ -592,6 +592,11 @@ def callback_mix(premix, snr=0, sir=0, ref_mic=0, n_src=None, n_tgt=None):
                 "RT60 between the {}th mic and {}th source: {:.3f} s".format(m, s, rt60[m, s])
             )
 
+Free-field simulation
+=====================
+
+You can also use this package to simulate free-field sound propagation between a set of sound sources and a set of microphones, without considering room effects. To this end, you can use the :py:obj:`pyroomacoustics.room.AnechoicRoom` class, which simply corresponds to setting the maximum image image order of the room simulation to zero. This allows for early development and testing of various audio-based algorithms, without worrying about room acoustics at first. Thanks to the modular framework of pyroomacoustics, room acoustics can easily be added, after this first testing stage, for more realistic simulations. 
+
```
Comment (fakufaku) [inline pyroomacoustics/room.py:599]: Can you please add a link to the relevant example in the paragraph ?

Also, a short code snippet would be awesome, but I leave that to you to decide! If you don't feel like it, don't do it 😄

File: docs/requirements.txt [hunk 5]
```diff
@@ -2,3 +2,4 @@ sphinx>=1.6.2
 mock>=2.0
 sphinx_rtd_theme
 sphinxcontrib-napoleon
+tabulate
```
Comment (fakufaku) [inline docs/requirements.txt:5]: Where is this necessary in the doc ?

File: docs/pyroomacoustics.bss.fastmnmf2.rst [hunk 6]
```diff
@@ -0,0 +1,7 @@
+pyroomacoustics.bss.fastmnmf2 module
```
Comment (fakufaku) [inline docs/pyroomacoustics.bss.fastmnmf2.rst:1]: I suppose this file and the two below were missing from the doc right ?

File: docs/requirements.txt (hunk 5 above)
Comment (fakufaku) [reply docs/requirements.txt:5]: Ah, the materials table, right ?

**Review by fakufaku — CHANGES_REQUESTED** [review]
Very cool, LGTM! Thanks, it makes it even easier to use in simple scenarios!

There were just a couple of things that I have added in comments. I can merge as soon as you are done.

**Review by duembgen — COMMENTED** [review]
Thank you for the review! 
I created an example script in the examples folder and added a corresponding code snippet to the docs. If you agree with all the changes you can merge this pull request.

File: pyroomacoustics/room.py (hunk 3 above)
Comment (duembgen) [reply pyroomacoustics/room.py:3006]: Thanks :)

**Comment by EliasBrohammer** [comment]
sorry for bursting into your conversation:
Just yesterday I created a script using the idea from issue #13 (max_order=0 for the room creation).
I will try the new room as soon as it's ready and give you feedback!

**Review by fakufaku — COMMENTED** [review]
Just a couple of things left!

**Comment by EliasBrohammer** [comment]
Just wanted to ask, how the status on this pull-request is? :)

**Comment by duembgen** [comment]
@fakufaku I have simplified the example, I hope that's roughly what you had in mind. Feel free to change anything to your liking and merge as soon as you are satisfied. 
@EliasBrohammer sorry for the late reply.

**Pull request merged**